rancid-tomatillos/src/App.js

```diff
@@ -1,30 +1,54 @@
 import React, {Component} from "react";
+import {BrowserRouter as Router, Switch, Route, Link} from "react-router-dom";
 import MainView from "./components/MainView";
-import CardsContainer from "./components/CardsContainer";
+import Header from "./components/Header";
+import Login from "./components/Login";
+import ShowPage from "./components/ShowPage";
 
 import "./App.css";
 
 class App extends Component {
     constructor() {
         super();
+        this.setLoggedIn = this.setLoggedIn.bind(this);
```

Comment:
`bind` was a key piece to sending the correct functionality to the child component (`Login`) - to alter state for `App` instead of its own state (or functional scope `this`? still hazy on that)

Reply:
Yeah I have a few ideas of how to mess around with this without binding, we will try them out!